packages/ra-core/src/controller/button/useDeleteWithConfirmController.tsx

```diff
@@ -80,18 +80,19 @@ const useDeleteWithConfirmController = ({
     const [deleteOne, { loading }] = useDelete(resource, null, null, {
         action: CRUD_DELETE,
         onSuccess: () => {
+            setOpen(false);
```

Comment:
A suggestion: It could be the occasion to add an `onEnd` event that would fire on both success and failure.
Or maybe it is overkill.

Reply:
It's not a little update because it should be implemented in all the hooks to keep the API consistent. Moreover, I'm not sure it's something we want in React-Admin. You can open an issue to discuss it :wink:.